AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -56,82 +59,100 @@ public final class APIClient: APIClientProtocol {
         )
     }
     
+    /// :nodoc:
+    @available(iOS 15.0.0, *)
+    public func perform<R: Request>(_ request: R) async -> Result<R.ResponseType, Error> {
+        do {
+            let result = try await urlSession.data(for: try buildUrlRequest(from: request))
+            
+            guard let httpResponse = result.1 as? HTTPURLResponse else {
+                fatalError("Invalid response.")
```

Comment:
this should be an error, not fatal

Reply:
yes you're right.

its just fatalError will crash the merchant app, should we crash merchant apps?. maybe create an error enum:

```Swift
public enum APIClientError: LocalizedError {
     case invalidResponse

    var errorDescription: String {
          return "Invalid Response" 
}
}
```